From 5159286aabc33e392ff4f0abb55f293812b74fe9 Mon Sep 17 00:00:00 2001
From: Jonathan DE CESCO <jonathan.de.cesco@intel.com>
Date: Mon, 13 May 2013 18:25:00 +0200
Subject: [PORT FROM MAIN] wifi: restore original SoftAP controller behavior

BZ: 107357

During bring-up phase, some hacks were made to turn on and off
wifi interfaces during wifi to hotspot switch.
This is not needed anymore and was slowing down wifi ON/OFF
cycles.
Also remove 11n mode parameter as it's not taken into account anyway
if hw_mode is not specified. Also 4335 is 11ac so specifying 11n may
give less throughput

Feature   : Wlan
Category  : Improvement
Purpose   : AOSP+
Origin    : Upstream
Upstream-candidate: no

Change-Id: I7dd3d61081fe3addc035a503948b57b647ea4b7d
Orig-Change-Id: I72e19a96b52bd4c3c0492964e08a9db14ef09187
Signed-off-by: Jonathan DE CESCO <jonathan.de.cesco@intel.com>
---
 SoftapController.cpp |   13 +------------
 1 files changed, 1 insertions(+), 12 deletions(-)

diff --git a/SoftapController.cpp b/SoftapController.cpp
index 60eb6ab..fcdf219 100644
--- a/SoftapController.cpp
+++ b/SoftapController.cpp
@@ -166,8 +166,7 @@ int SoftapController::setSoftap(int argc, char *argv[]) {
     ap_channel = atoi(ap_channel_s);
 
     asprintf(&wbuf, "interface=%s\ndriver=nl80211\nctrl_interface="
-            "/data/misc/wifi/hostapd\nssid=%s\nchannel=%d\nieee80211n=1\n"
-            "hw_mode=g\n",
+            "/data/misc/wifi/hostapd\nssid=%s\nchannel=%d\n",
 	     argv[2], argv[3], ap_channel);
 
     if (argc > 4) {
@@ -235,9 +234,6 @@ int SoftapController::fwReloadSoftap(int argc, char *argv[])
         return ResponseCode::CommandSyntaxError;
     }
 
-    iface = argv[2];
-    stopDriver(iface);
-
     if (!strcmp("STA", argv[3]))
         wifi_switch_driver_mode(WIFI_STA_MODE); /* which is STA + P2P... */
     else if (!strcmp("AP", argv[3]))
@@ -245,13 +241,6 @@ int SoftapController::fwReloadSoftap(int argc, char *argv[])
     else if (!strcmp("P2P", argv[3]))
         wifi_switch_driver_mode(WIFI_P2P_MODE);
 
-    /**
-     * Sleep to workaround issue in the brcm driver which is tracked by BZ
-     * 85864. To be removed as soon as BZ85864 is fixed.
-     */
-    sleep(2);
-    startDriver(iface);
-
     ALOGD("Softap fwReload - done");
 
     return ResponseCode::SoftapStatusResult;
-- 
1.7.4.1

